packages/react/src/runtime/__tests__/ac.test.ts

```diff
@@ -0,0 +1,116 @@
+import ac, { clearCache, getCache } from '../ac';
+
+describe('ac', () => {
+  const isEnabled: boolean = (() => false)();
+
+  beforeEach(() => {
+    clearCache();
+  });
+
+  it.each([
+    ['should handle empty array', [], undefined],
+    ['should handle array with undefined', [undefined], undefined],
+    ['should join single classes together', ['foo', 'bar'], 'foo bar'],
+    ['should join multi classes together', ['foo baz', 'bar'], 'foo baz bar'],
+    ['should remove undefined', ['foo', 'bar', undefined], 'foo bar'],
+    [
+      'should ensure the last atomic declaration of a single group wins',
+      ['_aaaabbbb', '_aaaacccc'],
+      '_aaaacccc',
+    ],
+    [
+      'should ensure the last atomic declaration of a single group with short class name wins',
+      ['_aaaabbbb', '_aaaacccc', '_aaaa_a'],
+      'a',
+    ],
+    [
+      'should ensure the last atomic declaration of many single groups wins',
+      ['_aaaabbbb', '_aaaacccc', '_aaaadddd', '_aaaaeeee'],
+      '_aaaaeeee',
+    ],
+    [
+      'should ensure the last atomic declaration of many single groups with short class name wins',
+      ['_aaaabbbb', '_aaaacccc', '_aaaa_a', '_aaaa_b'],
+      'b',
+    ],
+    [
+      'should ensure the last atomic declaration of a multi group wins',
+      ['_aaaabbbb _aaaacccc'],
+      '_aaaacccc',
+    ],
+    [
+      'should ensure the last atomic declaration of a multi group with short class name wins',
+      ['_aaaa_e', '_aaaabbbb _aaaacccc'],
+      '_aaaacccc',
+    ],
+    [
+      'should ensure the last atomic declaration of many multi groups wins',
+      ['_aaaabbbb _aaaacccc _aaaadddd _aaaaeeee'],
+      '_aaaaeeee',
+    ],
+    [
+      'should ensure the last atomic declaration of many multi groups with short class name wins',
+      ['_aaaabbbb', '_aaaa_a', '_bbbb_b', '_ddddcccc'],
+      'a b _ddddcccc',
+    ],
+    [
+      'should not remove any atomic declarations if there are no duplicate groups',
+      ['_aaaabbbb', '_bbbbcccc'],
+      '_aaaabbbb _bbbbcccc',
+    ],
+    [
+      'should not remove any atomic declarations if there are short class name and no duplicate groups',
+      ['_eeee_e', '_aaaabbbb', '_bbbbcccc'],
+      'e _aaaabbbb _bbbbcccc',
+    ],
+    ['should not apply conditional class', [isEnabled && 'foo', 'bar'], 'bar'],
```

Comment:
Having a bit of trouble understanding this test case - what would be an example where we might use `&&` and `||` operators in the input to `ac()`, and why would `isEnabled && 'foo'` not be part of `ac()`'s output?

Reply:
example:

```
import { styled } from '@compiled/react';

const MyDiv = styled.div({
  color: ({ isActive }) =>  isActive ? 'red' : undefined
});
```

which gets compiled to 
```
   const _ = "._syaz5scu{color:red}";
    const MyDiv = forwardRef(
      ({ as: C = "div", style: __cmpls, ...__cmplp }, __cmplr) => {
        const { isActive, ...__cmpldp } = __cmplp;
        return (
          <CC>
            <CS>{[_]}</CS>
            <C
              {...__cmpldp}
              style={__cmpls}
              ref={__cmplr}
              className={ac([
                __cmplp.isActive && "_syaz5scu",
                __cmplp.className,
              ])}
            />
          </CC>
        );
      }
    );
```

`ac([__cmplp.isActive && "_syaz5scu",__cmplp.className,])` would be an example where we might use `&&`.

As for `why would isEnabled && 'foo' not be part of ac()'s output`, I don't understand why you mean. Would you mind elaborating it?

Reply:
Thank you for the detailed explanation! :)

> As for why would isEnabled && 'foo' not be part of ac()'s output, sorry, I don't follow it. Would you mind elaborating it?

Ah sorry I wasn't very clear - I was wondering why the output would be just `'bar'` and not something like `'foo bar'` (or some other output that would incorporate `'foo'` somehow)

Reply:
oh, because of the line [here](https://github.com/atlassian-labs/compiled/pull/1437/files/fc90687444a03ceb28dec0fc30cd222f7371a39d#diff-53e489e609eaf51087b24f1e67079bc574f2cc03404278dccc22227241b2354fR4) , isEnabled is bound to return false in this test.